### DISCUSSION TITLE
Roles issues server RKE

### DISCUSSION THREAD

**Discussion by Davidstos** [rancher/rke2#4529] [Q&A] [answered]
**Environmental Info:**
RKE2 Version: 
v1.25.12+rke2r1

Node(s) CPU architecture, OS, and Version: 
OS: 22.04.2 LTS (Jammy Jellyfish)
CPU : x64 intel x86

Cluster Configuration:
"3 servers"

# kubectl get nodes
NAME      STATUS   ROLES                                   AGE   VERSION
vm22022   Ready    control-plane,etcd,master               48m   v1.25.12+rke2r1
vm22023   Ready    control-plane,etcd,master                33m   v1.25.12+rke2r1
vm22024   Ready    control-plane,etcd,master               27m   v1.25.12+rke2r1

**Describe the bug:**
Hello,

I have set up three rke2 servers, I don't want these servers to execute pods (disable the worker role). 
When I launch a deployment, the pods are mounted on my masters. I didn't find in the documentation how to disable the execution of pods on servers

NAMESPACE     NAME                                                    READY   STATUS      RESTARTS   AGE   IP             NODE      NOMINATED NODE   READINESS GATES
default       nginx-deployment-7fb96c846b-b55nf                       1/1     Running     0          5s    10.42.1.8      vm22023   <none>           <none>
default       nginx-deployment-7fb96c846b-fdzkl                       1/1     Running     0          5s    10.42.2.6      vm22024   <none>           <none>
default       nginx-deployment-7fb96c846b-p6tl5                       1/1     Running     0          5s    10.42.0.16     vm22022   <none>           <none>

**Comment by brandond** [comment, marked as answer]
You would want to use taints to do that:
https://docs.rke2.io/advanced#node-labels-and-taints

Ensure that you use a taint that the core components tolerate, otherwise the cluster will not come up completely without agent nodes.

**Reply by Davidstos** [reply]
Thank for your reply.

**Reply by brandond** [reply]
The most common taint used on control-plane nodes would be `node-role.kubernetes.io/control-plane=:NoSchedule`

**Reply by Davidstos** [reply]
Thank you, I found this in the doc `CriticalAddonsOnly=true:NoExecute` this seems to work too 🙂

**Reply by brandond** [reply]
that's another good one
